### PR TITLE
Rebase and Fix existing ref tests

### PR DESCRIPTION
**Pull request by chrisduerr**
This branch has been rebased on top of the latest master. I'm commiting this to neon64/alacritty so the scrollback PR is automatically updated.

Ref tests were broken because it assumed for existing tests that a
scrollback buffer is present. To fix this a check for tests without
scrollback has been added. Once a ref without scrollback is detected,
only the visible buffer is compared.

### CODE REVIEW

**Comment by chrisduerr** [comment]
Seems like I can't really resolve conflicts myself, or at least if I do it through github it doesn't work properly. But overriding everything with my branch will work and is probably a simple and quick fix.

**Comment by neon64** [comment]
Wouldn't I need to rebase my own fork first, and then apply your changes to ensure a clean commit history?

Thanks heaps for the work though by the way. I'll get around to it this afternoon.

**Comment by chrisduerr** [comment]
I went through the process again and thankfully this time github didn't decide to screw me over. So now everything should be good.

**Comment by neon64** [comment]
Hmm I've now rebased my own branch on top of jwilm:master, but for some reason GitHub doesn't like it and won't compared changes properly...

The bulk of the changes you've made are just in the commit `Fix existing ref tests` (f1a514b), yeah? Any way I can just merge that commit and leave the rest of the history intact?

**Comment by chrisduerr** [comment]
I've fixed the merge conflicts so there shouldn't have been a need to rebase your commit on top of the latest master. My version was already up to date with master. I'm not sure if your version will work, but I can resolve the conflicts again if that helps.

**Comment by chrisduerr** [comment]
Okay, @neon64 I've now resolved all conflicts again. Even tho it took me two go's until github was ready. Everything should be good to merge now.

File: src/config.rs [hunk 1]
```diff
@@ -1289,6 +1294,38 @@ impl Dimensions {
     }
 }
 
+/// Pixels per inch
+///
+/// This is only used on `FreeType` systems
+#[derive(Debug, Deserialize)]
+pub struct Dpi {
```
Comment (neon64) [inline src/config.rs]: This doesn't appear to be used anywhere else in the codebase. I think it was removed in https://github.com/jwilm/alacritty/commit/2ea20f4823ae96f92f47a1984a6dd118a9632fdb..

File: src/event.rs [hunk 2]
```diff
@@ -122,6 +122,13 @@ impl<'a, N: Notify + 'a> input::ActionContext for ActionContext<'a, N> {
         }
     }
 
+    fn visible_to_absolute(&self, point: Point) -> AbsolutePoint {
```
Comment (neon64) [inline src/event.rs]: This method is duplicated

**Review by neon64 — COMMENTED** [review]
Just found these two things that I don't think ought to be there

**Comment by chrisduerr** [comment]
Oh yeah, sorry for these mistakes. Looks like I tested master instead of my PR when adding these two merges. Now everything should be fixed and all test have completed successfully.

**Comment by neon64** [comment]
Okay I've merged it now.. Its sort of made the commit history a bit wacky - idk what's going on but I hope its okay. Thanks heaps for the input!

**Pull request merged**